indexer delegated staking (#156)
aptos-labs/aptos-docs: src/content/docs/build/indexer/indexer-api/indexer-reference.mdx

| cufab\_owner\_at\_index                         | owner\_address, asset\_type |
| current\_unified\_fungible\_asset\_balances\_pkey | storage\_id                |

## Delegated Staking

> 

Reviewer: ```suggestion

With [AIP-6](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-6.md) we added the ability for delegation pools to be created. Once these pools hold over 1M APT, they can be used as a validator pool.

```

### `current_delegated_staking_pool_balances`

This table tracks the current balances of each account in a delegated staking pool.

| Field                        | Type   | Primary Key | Description                                                                 |
|------------------------------|--------|-------------|-----------------------------------------------------------------------------|

> | staking\_pool\_address         | String | Yes         | The address of the staking pool.                                            |

Reviewer: ```suggestion
| staking\_pool\_address         | String | Yes         | The address of the delegation pool.
```

| total\_coins                  | bigint |             | The total number of coins in the staking pool.                              |

> | total\_shares                 | bigint |             | The total number of shares in the staking pool.                             |

Reviewer: ```suggestion
| total\_shares                 | bigint |             | The total number of shares in the delegation pool. |
```

> | last\_transaction\_version     | int8   |             | Blockchain version of the last transaction involving this staking pool. Ex. 30000000 |

Reviewer: ```suggestion
| last\_transaction\_version     | int8   |             | Transaction version (identifier) for the last transaction involving this staking pool. |
```

> | inserted\_at                  | String |             | The timestamp when the record was inserted.                                 |

Reviewer: i would move this to the end

> | operator\_commission\_percentage | bigint |          | The commission percentage taken by the operator.                            |

Reviewer: ```suggestion
| operator\_commission\_percentage | bigint |          | The commission percentage taken by the staking pool operator.                 |
```

| inactive\_table\_handle        | String |             | The table handle for the inactive table.                                    |
| active\_table\_handle          | String |             | The table handle for the active table.                                      |

| Index Name | Indexed Columns |
|------------|----------------|
| current\_delegated\_staking\_pool\_balances\_pkey | staking\_pool\_address |

### `current_delegated_voter`

This table tracks the current delegated voters of a delegation pool.

| Field                      | Type   | Primary Key | Description |
|----------------------------|--------|-------------|-------------|
| delegation\_pool\_address    | String | Yes         | The address of the delegation pool. |
| delegator\_address          | String | Yes         | The address of the delegator. |

> | table\_handle               | String |             | The table handle for the delegation pool. |

Reviewer: ```suggestion
| table\_handle               | String |             | The table handle tracking this position. |
```

| voter                      | String |             | The address of the current voter in the delegation pool. |
| pending\_voter              | String |             | The address of the pending voter awaiting confirmation. |

> | last\_transaction\_version   | bigint |             | The version number of the last transaction involving this delegation. |

Reviewer: ```suggestion
| last\_transaction\_version   | bigint |             | The transaction version (identifier) of the last transaction involving this delegation. |
| last\_transaction\_timestamp | Timestamp |          | The block timestamp of the last transaction involving this delegation. |
```

> | last\_transaction\_version   | bigint |             | The version number of the last transaction involving this delegation. |
> | last\_transaction\_timestamp | Timestamp |          | The timestamp of the last transaction involving this delegation. |

Reviewer: ```suggestion
| last\_transaction\_version   | bigint |             | The transaction version (identifier) of the last transaction involving this delegation. |
| last\_transaction\_timestamp | Timestamp |          | The block timestamp of the last transaction involving this delegation. |
```

| inserted\_at                | Timestamp |          | The timestamp when the record was inserted into the database. |

| Index Name | Indexed Columns |
|------------|----------------|
| current\_delegated\_voter\_pkey | delegation\_pool\_address, delegator\_address |
| cdv\_da\_index | delegator\_address |

### `current_delegator_balances`

This table tracks the current balances of each account in a delegated staking pool.

| Field                    | Type   | Primary Key | Description |
|----------------------------|--------|-------------|-------------|
| delegator\_address          | String | Yes            | The address of the delegator.            |
| pool\_address               | String |             | The address of the delegator pool.            |

> | pool\_type                | String |             | The type of the pool. Ex. "delegated" |

Reviewer: ```suggestion
| pool\_type                | String |             | If the shares are active or inactive |
```

| table\_handle             | String |             | The table handle for the pool. |
| last\_transaction\_version | bigint |             | The version number of the last transaction involving this pool. Ex. 5000000 |

> | inserted\_at              | Timestamp |             | The timestamp when the record was inserted. |

Reviewer: move to the end

| shares                   | bigint |             | The number of shares in the pool. |
| parent\_table\_handle      | String |             | The table handle for the parent table. |

| Index Name | Indexed Columns |
|------------|----------------|
| current\_delegator\_balances\_pkey | delegator\_address, pool\_address, pool\_type, table\_handle |

### `current_staking_pool_voter`

This table tracks the current voters of a staking pool.

| Field                    | Type   | Primary Key | Description |
|----------------------------|--------|-------------|-------------|
| staking\_pool\_address       | String | Yes            | The address of the staking pool.            |
| voter\_address              | String |             | The address of the voter.            |

> | last\_transaction\_version | bigint |             | The version number of the last transaction involving this pool. Ex. 5000000 |

Reviewer: ```suggestion
| last\_transaction\_version     | bigint   |             | Transaction version (identifier) for the last transaction involving this staking pool. |
```

> | inserted\_at              | Timestamp |             | The timestamp when the record was inserted. |

Reviewer: move to end

| operator\_address         | String |             | The address of the operator. |

| Index Name | Indexed Columns |
|------------|----------------|
| current\_staking\_pool\_voter\_pkey | staking\_pool\_address |
| ctpv\_va\_index | voter\_address |
| ctpv\_insat\_index | inserted\_at |

### `delegated_staking_activities`

This table tracks delegated staking events.

| Field                    | Type   | Primary Key | Description |
|----------------------------|--------|-------------|-------------|
| transaction\_version      | bigint |             | The version number of the transaction. Ex. 5000000 |
| event\_index              | bigint |             | The index of the event. Ex. 1 |
| delegator\_address        | String |             | The address of the delegator.            |
| pool\_address             | String |             | The address of the pool.            |

> | event\_type               | String |             | The event type. Ex. "0x1::stake::DistributeRewardsEvent" |

Reviewer: ```suggestion
| event\_type               | String |             | DistributeRewards, AddStake, UnlikeStake, ReactiveStake, WithdrawStake |
```

| amount                   | bigint |             | The amount being staked. Ex. 1000 |
| inserted\_at              | Timestamp |             | The timestamp when the record was inserted. |

| Index Name | Indexed Columns |
|------------|----------------|
| delegated\_staking\_activities\_pkey | transaction\_version, event\_index |
| dsa\_pa\_da\_index | pool\_address, delegator\_address, transaction\_version, event\_index |
| dsa\_insat\_index | pool\_address |

### `delegated_staking_pool_balances`

This table tracks the historical balances of each account in a delegated staking pool.

> | Field                    | Type   | Primary Key | Description |

Reviewer: same change sas above

|----------------------------|--------|-------------|-------------|
| transaction\_version        | bigint |             | The version number of the transaction. Ex. 5000000 |
| staking\_pool\_address       | String |             | The address of the staking pool. |
| total\_coins                  | bigint |             | The total number of coins in the staking pool. |
| total\_shares                 | bigint |             | The total number of shares in the staking pool. |
| inserted\_at                  | String |             | The timestamp when the record was inserted. |
| operator\_commission\_percentage | bigint |             | The commission percentage taken by the operator. |
| inactive\_table\_handle        | String |             | The table handle for the inactive table. |
| active\_table\_handle          | String |             | The table handle for the active table. |

| Index Name | Indexed Columns |
|------------|----------------|
| delegated\_staking\_pool\_balances\_pkey | transaction\_version, staking\_pool\_address |

> ### `delegated_staking_pools`

Reviewer: description is fine but not sure why you'd expect to only have tx_version when querying for this table, not sure this table is needed

> This table tracks the first transaction version that a pool was created.

Reviewer: ```suggestion
This table when a delegated pool was created.
```
